lib/services/cordova-project-plugins-service.ts

```diff
@@ -0,0 +1,682 @@
+	private static CORDOVA_PLUGIN_VARIABLES_PROPERTY_NAME = "CordovaPluginVariables";
+	private static MESSAGES = ["Core Plugins", "Advanced Plugins", "Marketplace Plugins"];
+	private _identifierToPlugin: IDictionary<IPlugin>;
+	private get identifierToPlugin(): IDictionary<IPlugin> {
+		if (!this._identifierToPlugin) {
+			this.loadPluginsData().wait();
+		}
+
+		return this._identifierToPlugin;
+	}
+
+	private pluginsForbiddenConfigurations: IStringDictionary = {
+		"com.telerik.LivePatch": this.$projectConstants.DEBUG_CONFIGURATION_NAME
+	};
+
+	constructor(private $cordovaPluginsService: cordovaPlugins.CordovaPluginsService,
+		private $marketplacePluginsService: ICordovaPluginsService,
+		private $errors: IErrors,
+		private $logger: ILogger,
+		private $project: Project.IProject,
+		private $prompter: IPrompter,
+		private $loginManager: ILoginManager,
+		private $projectConstants: Project.IProjectConstants,
+		private $options: IOptions,
+		private $fs: IFileSystem) { }
+
+	private loadPluginsData(): IFuture<void> {
+		return (() => {
+			// Cordova plugin commands are only applicable to Cordova projects
+			this.$project.ensureCordovaProject();
+			this.$loginManager.ensureLoggedIn().wait();
+			this._identifierToPlugin = Object.create(null);
+			Future.wait([this.createPluginsData(this.$cordovaPluginsService),
+				this.createPluginsData(this.$marketplacePluginsService)]);
+		}).future<void>()();
+	}
+
+	public findPlugins(keywords: string[]): IFuture<IBasicPluginInformation[]> {
+		return (() => {
+			return this.$cordovaPluginsService.getPlugins(keywords);
+		}).future<IBasicPluginInformation[]>()();
+	}
+
+	public fetch(pluginIdentifiers: string[]): IFuture<void> {
+		return (() => {
+			if (pluginIdentifiers.length === 0) {
+				this.$logger.error("You must specify local path, URL to a plugin repository, name or keywords of a plugin published to the Cordova Plugin Registry.");
+			} else if (pluginIdentifiers.length === 1 && (this.isLocalPath(pluginIdentifiers[0]).wait() || this.isUrlToRepository(pluginIdentifiers[0]))) {
+				let result = this.$cordovaPluginsService.fetch(pluginIdentifiers[0]).wait();
+				this.$logger.out(result);
+			} else {
+				let plugins = this.$cordovaPluginsService.getPlugins(pluginIdentifiers);
+				let pluginKeys = Object.keys(plugins);
+				let pluginsCount = pluginKeys.length;
+				if (pluginsCount === 0) {
+					this.$logger.out("There are 0 matching plugins.");
+				} else if (pluginsCount > 1 && pluginKeys[0] !== pluginIdentifiers[0]) {
+					this.$logger.out("There are more then 1 matching plugins.");
+				} else {
+					this.$logger.out(this.$cordovaPluginsService.fetch(pluginKeys[0]).wait());
+				}
+			}
+		}).future<void>()();
+	}
+
+	private isLocalPath(pluginId: string): IFuture<boolean> {
+		return this.$fs.exists(pluginId);
+	}
+
+	private isUrlToRepository(pluginId: string): boolean {
+		return validUrl.isUri(pluginId);
+	}
+
+	public getInstalledPlugins(): IPlugin[] {
```

Comment:
You can consider passing `debug` and `release` as `boolean` parameters instead of resolving them from `this.$options`